pytorch_lightning/trainer/trainer.py

```diff
@@ -2004,6 +2004,10 @@ def _results(self) -> Optional[ResultCollection]:
         if active_loop is not None:
             return active_loop._results
 
+    def _exit_gracefully_on_signal(self, status: str) -> None:
+        if _fault_tolerant_training() and self._terminate_gracefully:
+            raise ExitGracefullyException(f"Exiting on {status}")
```

Comment:
is this expected to be thrown during any part of the trainer execution? 
what's required for the error handling here? https://github.com/PyTorchLightning/pytorch-lightning/blob/019e69ffd2d4e1f86f9002901cf217f85660b4b1/pytorch_lightning/trainer/trainer.py#L558-L567

Reply:
Nothing to be changed on the exception handling part. 
It won't be thrown anywhere, just in safe places where all components have been properly exectuted.
It would be raised only if `self._terminate_gracefully` has been set to True through a signal notification.